cache-config/testing/ort-tests/t3c-ims_test.go

```diff
@@ -27,158 +26,150 @@ import (
+	if err != nil {
+		t.Fatalf("getting profile: %v", err)
+	}
+	if len(cdns) != 1 {
+		t.Fatalf("getting profile: expected 1 cdn actual num %d objects: %+v", len(profiles), profiles)
+	}
 
-		remapName := filepath.Join(test_config_dir, "remap.config")
-		remapDotConfig, err := ioutil.ReadFile(remapName)
-		if err != nil {
-			t.Fatalf("reading %v: %v\n", remapName, err)
-		}
+	cdn2ID := cdns[0].ID
+	cdn2ProfileID := profiles[0].ID
 
-		if !bytes.Contains(remapDotConfig, []byte(cdn1Domain)) {
-			t.Errorf("expected remap.config to contain cdn1 domain '%v', actual: '%v'\n", cdn1Domain, string(remapDotConfig))
-		}
+	sv, _, err := GetServer(tcdata.TOSession, cacheHostName)
+	if err != nil {
+		t.Fatalf("getting server: %v", err)
 	}
 
-	{
-		// change the server's CDN
-
-		cdn2Name := "cdn2"
-		cdns, _, err := tcdata.TOSession.GetCDNByNameWithHdr(cdn2Name, nil)
-		if err != nil {
-			t.Fatalf("getting cdn: " + err.Error())
-		} else if len(cdns) != 1 {
-			t.Fatalf("getting cdn: expected 1 cdn actual num %v cdns %+v", len(cdns), cdns)
-		}
+	sv.CDNID = &cdn2ID
+	sv.CDNName = &cdn2Name
+	sv.ProfileID = &cdn2ProfileID
+	sv.Profile = util.StrPtr(cdn2ProfileName)
 
-		// have to change the profile at the same time, or TO will reject the change.
-		profiles, _, err := tcdata.TOSession.GetProfileByNameWithHdr(cdn2ProfileName, nil)
-		if err != nil {
-			t.Fatalf("getting profile: " + err.Error())
-		} else if len(cdns) != 1 {
-			t.Fatalf("getting profile: expected 1 cdn actual num %v objects %+v", len(profiles), profiles)
-		}
+	_, _, err = tcdata.TOSession.UpdateServerByIDWithHdr(*sv.ID, *sv, nil)
+	if err != nil {
+		t.Fatalf("updating server: %v", err)
+	}
+}
 
-		cdn2ID := cdns[0].ID
-		cdn2ProfileID := profiles[0].ID
+func checkChangedCDN(t *testing.T) {
+	remapName := filepath.Join(test_config_dir, "remap.config")
+	remapDotConfig, err := ioutil.ReadFile(remapName)
+	if err != nil {
+		t.Fatalf("reading %s: %v", remapName, err)
+	}
+	contents := string(remapDotConfig)
 
-		sv, _, err := GetServer(tcdata.TOSession, cacheHostName)
-		if err != nil {
-			t.Fatalf("getting server: " + err.Error())
-		}
+	if !strings.Contains(contents, cdn2Domain) {
+		t.Errorf("expected after changing server to cdn2 for remap.config to contain cdn2 domain '%s', actual: '%s'", cdn2Domain, contents)
+	}
 
-		sv.CDNID = &cdn2ID
-		sv.CDNName = &cdn2Name
-		sv.ProfileID = &cdn2ProfileID
-		sv.Profile = &cdn2ProfileName
+	if strings.Contains(contents, cdn1Domain) {
+		t.Errorf("expected after changing server to cdn2 for remap.config to not contain cdn1 domain '%s', actual: '%s'", cdn1Domain, remapDotConfig)
+	}
+}
 
-		_, _, err = tcdata.TOSession.UpdateServerByIDWithHdr(*sv.ID, *sv, nil)
-		if err != nil {
-			t.Fatalf("updating server: " + err.Error())
-		}
+// doTestIMSChangedCDN tests that after caching, requests which use the CDN as a key don't use the invalid cache.
+func doTestIMSChangedCDN(t *testing.T) {
```

Comment:
As previously, I found it really hard to understand the components of this test. It breaks the test into smaller functions, which is great. But what those functions do was very unexpected, at least to me.

Things like `t.Run(checkRemapConfigHasInitialCDN)` that isn't a full test, and depends on setup in the external function. But it also isn't even the test itself, it's just verifying the server had cdn1, because testing the change would be a lie if it were already on cdn2. It's really a pre-test validation.

Likewise, `t.Run(changeServerCDN` is just setup, not really a test, and then we run t3c again in the outside func, and `t.Run(checkChangedCDN` is the real test, but doesn't include its own setup.

Considering the `checkRemapConfigHasInitialCDN` only exists to verify the real test is valid, I think this whole func is really one test. Even though it's kind of big, `t.Run` sub-tests don't really seem to make sense.

Reply:
That works for me. The purpose of `t.Run` calls was to eliminate anonymous block scopes, but I have no problem simply collapsing those into the scope of `doTestIMSChangedCDN`.